lib/dangermattic/plugins/manifest_pr_checker.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Danger
+  # Plugin to check if the Gemfile.lock was updated when changing the Gemfile in a PR.
+  class ManifestPRChecker < Plugin
+    # Performs all the checks, asserting that changes on `Gemfile`, `Podfile` and `Package.swift` must have corresponding
+    # lock file changes.
+    def check_all_manifest_lock_updated
+      check_gemfile_lock_updated
+      check_podfile_lock_updated
+      check_swift_package_resolved_updated
+    end
+
+    # Check if the `Gemfile` file was modified without a corresponding `Gemfile.lock` update
+    def check_gemfile_lock_updated
+      check_manifest_lock_updated(
+        file_name: 'Gemfile',
+        lock_file_name: 'Gemfile.lock',
+        instruction: 'Please run `bundle install` or `bundle update <updated_gem>`'
+      )
+    end
+
+    # Check if the `Podfile` file was modified without a corresponding `Podfile.lock` update
+    def check_podfile_lock_updated
+      check_manifest_lock_updated(
+        file_name: 'Podfile',
+        lock_file_name: 'Podfile.lock',
+        instruction: 'Please run `bundle exec pod install`'
+      )
+    end
+
+    # Check if the `Package.swift` file was modified without a corresponding `Package.resolved` update
+    def check_swift_package_resolved_updated
+      check_manifest_lock_updated(
+        file_name: 'Package.swift',
+        lock_file_name: 'Package.resolved',
+        instruction: 'Please resolve the Swift packages in Xcode'
+      )
+    end
+
+    private
+
+    def check_manifest_lock_updated(file_name:, lock_file_name:, instruction:)
+      manifest_modified_file = git.modified_files.find { |f| f.end_with?(file_name) }
```

Comment:
This only finds the first manifest file.

<details><summary>What if there are many of them (like for the case of <code>Components/**/Podfile</code> in Tumblr-iOS)?</summary>
<img width="517" alt="image" src="https://github.com/Automattic/dangermattic/assets/216089/234186f7-ff85-46df-b67d-87aba8950d71">
</details>

Reply:
You should check `File.basename(f) == file_name` instead of `f.end_with?(file_name)`. Because, even if it's very unlikely that such a file will exist in the repo, `end_with?('Podfile')` would also pick up `Sample/ExamplePodfile` and similar, not just files whose basename is exactly `Podfile`.